### PR TITLE
Add back bitswap metrics and global cid blacklist

### PR DESCRIPTION
**Pull request by kylehuntsman**
Adds back the missing bitswap metrics and global cid blacklist functionality

### CODE REVIEW

**Comment by kylehuntsman** [comment]
@elijaharita Preferably, this PR should be rebased and merged instead of squashed.

File: bitswap/provider.go [hunk 1]
```diff
@@ -176,61 +176,41 @@ func (provider *Provider) ReceiveMessage(ctx context.Context, sender peer.ID, in
...
 			}
 
 			provider.blockManager.GetAwait(ctx, entry.Cid, func(block blocks.Block) {
-				stats.Record(context.Background(), metrics.BitswapResponseCount.M(1))
-				provider.taskQueue.PushTasks(sender, peertask.Task{
-					Topic:    topicBlock(block),
-					Priority: int(entry.Priority),
-					Work:     len(block.RawData()),
-				})
+				provider.queueBlock(ctx, sender, entry, block)
```
Comment (elijaharita) [inline bitswap/provider.go:213]: `ctx` gets passed into a a callback here, which is stored and invoked later. By the time that happens, `ctx` could already be closed, so for that reason I'd pass `context.Background()` here. But I'm not sure if that messes up the tag for stats recording in that function?

What do you think?

Comment (kylehuntsman) [reply bitswap/provider.go:213]: Ah, okay. Let me try this again with that in mind. My gut says using `context.Background()` wouldn't change anything. Thanks for the heads up.

**Review by elijaharita — APPROVED** [review]
Cool thanks, looks good! Clear to merge : )

**Pull request merged**